src/equations/hyperbolic_serre_green_naghdi_1d.jl

```diff
@@ -0,0 +1,364 @@
+@doc raw"""
+    HyperbolicSerreGreenNaghdiEquations1D(bathymetry_type = bathymetry_mild_slope;
+                                          gravity_constant, eta0 = 0.0, λ)
+
+Hyperbolic approximation of the Serre-Green-Naghdi system in one spatial
+dimension. The equations for flat bathymetry are given by
+```math
+\begin{aligned}
+  h_t + (h v)_x &= 0,\\
+  h v_t + \frac{1}{2} g (h^2)_x + \frac{1}{2} h (v^2)_x
+    + \biggl( \frac{\lambda}{3} \eta (1 - \eta / h) \biggr)_x &= 0,\\
+  h w_t + h v w_x &= \lambda (1 - \eta / h),\\
+  \eta_t + \eta_x u &= w.
+\end{aligned}
+\end{aligned}
+```
+The unknown quantities of the hyperbolized Serre-Green-Naghdi equations are the
+total water height ``\eta = h + b`` and the velocity ``v``.
+The gravitational constant is denoted by `g` and the bottom topography
+(bathymetry) ``b = \eta_0 - D``. The water height above the bathymetry
+is therefore given by ``h = \eta - \eta_0 + D``.
+The total water height is therefore given by ``\eta = h + b``.
+
+Two types of variable `bathymetry_type` are supported:
+- [`bathymetry_flat`](@ref): flat bathymetry (typically ``b = 0`` everywhere)
+- [`bathymetry_mild_slope`](@ref): variable bathymetry with mild-slope approximation
+
+For the mild-slope approximation, the Serre-Green-Naghdi equations are
```

Comment:
This system corresponds to a mild-slope approximation of the bathymetry? From the preprint I would have expected that this is for a general variable bathymetry and the mild-slope approximation was only applied to the full system. If this system indeed describes a mild-slope approximation, does there also exist a formulation for a general bathymetry without the mild-slope assumption?

Reply:
The papers of Gavrilyuk et al. that we looked at only developed an approximation for flat bathymetry (Favrie and Gavrilyuk) with mild-slope approximation. That's why we got a mild-slope version for the original SGN equations when translating the split form from the hyperbolic approximation.

Reply:
Ok, thanks for the clarification!